### PR TITLE
[WIP] Support Deepgrow 2D/3D Training

### PR DESCRIPTION
**Pull request by SachidanandAlle**
Fixes # .

### Description
Support Deegrow 2D/3D training (training example will be added into Tutorials)

### Status
**Work in progress**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
thanks! i put a few todos here for discussion:
- the transforms should inherit monai.transforms.Transform so that the dataset caching could work properly https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/compose.py#L26
- ideally the core functionality should only rely on numpy and torch, the other packages should be imported with the optional_import method https://github.com/Project-MONAI/MONAI/blob/4907f3908c3d222bb3b486495ddc3f892962be75/monai/utils/module.py#L114
- make_grid_with_titles looks nice, it could be in the visualize folder (is it possible to avoid cv2, and add support for https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_image)
- scipy.ndimage.filters.gaussian_filter could be replaced by https://github.com/Project-MONAI/MONAI/blob/4907f3908c3d222bb3b486495ddc3f892962be75/monai/networks/layers/simplelayers.py#L133 example: https://github.com/Project-MONAI/MONAI/blob/4f2df00ffb820e7a1ff0ceb71a079ca1bf8b547f/monai/data/utils.py#L522
- some normalisation could be replaced by monai.transforms.ScaleIntensityRange https://github.com/Project-MONAI/MONAI/blob/4907f3908c3d222bb3b486495ddc3f892962be75/monai/transforms/intensity/array.py#L282
- DeepgrowStatsHandler implements quite a few features might need another iteration to split it into smaller/specific event handlers.

I feel it's important to set up some regression tests before any of these refactorings it'll also be very helpful for other people that continue maintaining this module

cc @Nic-Ma

**Comment by tvercaut** [comment]
Interesting. I couldn't see a paper reference within the code. Is this a re-implementation of https://arxiv.org/abs/1903.08205?

Having some documentation, and maybe even an exemplar [tutorial](https://github.com/Project-MONAI/tutorials) / [example use case](https://github.com/Project-MONAI/research-contributions) would be great.

**Comment by SachidanandAlle** [comment]
Refer: https://github.com/Project-MONAI/MONAI/pull/1395
